### PR TITLE
refactor: Added name for background task started on enterBackground 

### PR DESCRIPTION
**Pull request by antohisorin**
### Summary

Added name for background task started on enterBackground to improve debugging.

### Checklist

* [X] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-iOS/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* [X] Does your PR have a breaking change?:  NO

### CODE REVIEW

**Comment by antohisorin** [comment]
@Mercy811 @qingzhuozhen is there a way we can land this MR? We need to track background tasks and what happens to them in our app. We don't want to fork the amplitude SDK only for this small change.

File: Sources/Amplitude/Amplitude.m [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ - (void)enterBackground {
 
     // Stop uploading
     [self endBackgroundTaskIfNeeded];
-    _uploadTaskID = [app beginBackgroundTaskWithExpirationHandler:^{
+    _uploadTaskID = [app _uploadTaskID = [app beginBackgroundTaskWithName:@"Amplitude: enterBackground" expirationHandler::^{
```
Comment (izaaz) [inline Sources/Amplitude/Amplitude.m:1253]: @antohisorin Looks like this does not compile? Shoudl this be

```
    _uploadTaskID = [app beginBackgroundTaskWithName:@"Amplitude: enterBackground" expirationHandler:^{
```

**Comment by izaaz** [comment]
> @Mercy811 @qingzhuozhen @justin-fiedler is there a way we can land this MR? We need to track background tasks and what happens to them in our app. We don't want to fork the amplitude SDK only for this small change.

Sure @antohisorin. I can help get this merged and release a new version. Dropped a comment

File: Sources/Amplitude/Amplitude.m (hunk 1 above)
Comment (antohisorin) [reply Sources/Amplitude/Amplitude.m]: Updated.

**Comment by antohisorin** [comment]
> > @Mercy811 @qingzhuozhen @justin-fiedler is there a way we can land this MR? We need to track background tasks and what happens to them in our app. We don't want to fork the amplitude SDK only for this small change.
> 
> Sure @antohisorin. I can help get this merged and release a new version. Dropped a comment

Updated. Sorry I don't know why I had duplicate values there.

**Review by izaaz — APPROVED** [review]

**Comment by izaaz** [comment]
Thanks @antohisorin . I'll get this merged and deployed once the tests pass.

**Pull request merged**